### ISSUE TITLE
Css/With binding bug

### ISSUE THREAD

**Issue by ghost** [knockout/knockout#579]
When the css: binding is used in a with: binding context, and the with binding contains a conditional, the css: binding fails to properly toggle classes. Instead, when removing a class, it removes all classes that were dynamically added after the model was bound. Here are some examples:

This is the expected behaviour - the added css class depends on an observable model variable. When it is removed, all the other classes remain there.
http://jsfiddle.net/47NBC/

However, when used in a conditional with: context, all the classes that were added after applyBindings() was called are removed:
http://jsfiddle.net/TgdEp/

It can be shown that the code works properly when the classes are added before the applyBindings() call:
http://jsfiddle.net/XscVu/

And the code also works when the with: is not a conditional binding:
http://jsfiddle.net/3rjM5/

So, to summarize (TLDR): when the with: binding contains a conditional statement, css: bindings within its scope tend to remove all dynamic classes added after applyBindings() was called.
